src/javascript/JContent/actions/createFolderAction/createFolderAction.jsx

```diff
@@ -1,21 +1,54 @@
-import React from 'react';
-import {composeActions, componentRendererAction} from '@jahia/react-material';
-import requirementsAction from '../requirementsAction';
+import React, {useContext} from 'react';
 import CreateFolderDialog from './CreateFolderDialog';
+import {useNodeChecks} from '@jahia/data-helper';
+import {ComponentRendererContext} from '@jahia/ui-extender';
+import PropTypes from 'prop-types';
 
-export default composeActions(requirementsAction, componentRendererAction, {
-    init: context => {
-        context.initRequirements({requiredPermission: 'jcr:addChildNodes'});
-    },
-    onClick: context => {
-        const onExit = () => {
-            handler.destroy();
-        };
-
-        let handler = context.renderComponent(
-            <CreateFolderDialog node={context.node}
-                                contentType={context.contentType}
-                                onExit={onExit}/>
-        );
+export const CreateFolderAction = ({context, render: Render, loading: Loading}) => {
+    const componentRenderer = useContext(ComponentRendererContext);
+    const res = useNodeChecks(
+        {path: context.path},
+        {requiredPermission: ['jcr:addChildNodes'], ...context}
+    );
+
+    if (res.loading && Loading) {
+        return <Loading context={context}/>;
     }
-});
+
+    const onExit = () => {
+        componentRenderer.destroy('createFolderDialog');
+    };
+
+    return (
+        <Render context={{
+            ...context,
+            isVisible: res.checksResult,
+            enabled: res.checksResult,
+            onClick: () => {
+                componentRenderer.render('createFolderDialog', Dialog, {path: context.path, contentType: context.contentType, onExit: onExit});
+            }
+        }}/>
+    );
+};
+
+const Dialog = ({path, contentType, onExit}) => {
```

Comment:
What is Dialog doing .. ? Can't we just remove it ?

Reply:
nothing, just needed the element to render it by componentRenderer

Reply:
ok so please remove it :-)